ee/tabby-webserver/src/service/auth.rs

```diff
@@ -218,6 +220,53 @@ impl AuthenticationService for AuthenticationServiceImpl {
         Ok(resp)
     }
 
+    async fn request_password_reset_email(&self, email: String) -> Result<String> {
+        let user = self.get_user_by_email(&email).await.ok();
+
+        let Some(user @ User { active: true, .. }) = user else {
+            return Ok(Uuid::new_v4().to_string());
+        };
+
+        let id = user.id.as_rowid()?;
+        let existing = self.db.get_password_reset_by_user_id(id).await?;
+        if let Some(existing) = existing {
+            if Utc::now().signed_duration_since(existing.created_at) < Duration::minutes(5) {
+                return Err(anyhow!(
+                    "A password reset has been requested recently, please try again later"
+                ));
+            }
+        }
+        let code = self.db.create_password_reset(id).await?;
+        self.mail
+            .send_password_reset_email(user.email, code.clone())
+            .await?;
+        Ok(code)
+    }
+
+    async fn password_reset(&self, code: &str, password: &str) -> Result<(), PasswordResetError> {
+        let password_encrypted =
+            password_hash(password).map_err(|_| PasswordResetError::Unknown)?;
+
+        let Ok(Some(password_reset)) = self.db.get_password_reset_by_code(code).await else {
+            return Err(PasswordResetError::InvalidCode);
+        };
+
+        let user_res = self.db.get_user(password_reset.user_id).await;
+        let Ok(Some(user @ UserDAO { active: true, .. })) = user_res else {
+            return Err(PasswordResetError::InvalidCode);
+        };
+
+        if Utc::now().signed_duration_since(password_reset.created_at) > Duration::minutes(15) {
+            return Err(PasswordResetError::InvalidCode);
+        }
```

Comment:
consider extract these into `fn verify_reset_password_code(code) -> i64` into `DbConn`, thus the expiration time related to password reset will be in single place (otherwise user won't know how long it become expire by looking at tabby-db/src/password_reset.rs.